### PR TITLE
Feature/pdct 1067 user org not checked on document create

### PR DESCRIPTION
**Pull request by katybaulch**
# Description

Check user org matches org the family we want to add the document to belongs to on document create.

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [x] Skip auto-tagging
- [ ] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [ ] Remove legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] The PR represents a single feature (small drive-by fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: tests/integration_tests/document/test_create.py [hunk 1]
```diff
@@ -46,8 +46,84 @@ def test_create_document(client: TestClient, data_db: Session, user_header_token
...
+    )
+    assert actual_pd is not None
+    assert actual_pd.title == "Title"
+
+    slug = (
+        data_db.query(Slug)
+        .filter(Slug.family_document_import_id == actual_fd.import_id)
+        .one()
+    )
+    assert len(slug.name) == len("title") + 1 + 4
```
Comment (diversemix) [inline tests/integration_tests/document/test_create.py:83]: Maybe a comment on the magic numbers?

File: tests/unit_tests/routers/document/test_create_document.py [hunk 2]
```diff
@@ -29,3 +29,46 @@ def test_create_when_family_not_found(
     data = response.json()
     assert data["detail"] == "Could not find family for this_family"
     assert document_service_mock.create.call_count == 1
+
+
+def test_create_fails_when_validation_error(
```
Comment (diversemix) [inline tests/unit_tests/routers/document/test_create_document.py:34]: Good addition :+1:

**Review by diversemix — APPROVED** [review]
On :fire: today @katybaulch :gem:

File: tests/integration_tests/document/test_create.py (hunk 1 above)
Comment (diversemix) [reply tests/integration_tests/document/test_create.py]: I think the comment should be # 4 = slug hash length 1 = separator

Comment (katybaulch) [reply tests/integration_tests/document/test_create.py]: Have added SLUG_SUFFIX_LENGTH to represent 4, but not sure what the one refers to..

**Review by diversemix — APPROVED** [review]
Better code always better than better comments - nice :1st_place_medal:

**Pull request merged**